crates/bevy_render/macros/src/atomic_id.rs

```diff
@@ -0,0 +1,47 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::parse::{Parse, ParseStream};
+use syn::{parse_macro_input, Ident, Visibility};
+
+pub struct AtomicIdStruct {
+    pub vis: Visibility,
+    pub ident: Ident,
+}
+
+impl Parse for AtomicIdStruct {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            vis: input.parse()?,
+            ident: input.parse()?,
+        })
+    }
+}
+
+pub fn define_atomic_id(input: TokenStream) -> TokenStream {
```

Comment:
This doesn't need to be a proc macro. It could easily be defined via `macro_rules`. In a crate that is already the compilation bottleneck for the engine in its current state, it's probably a good idea to add more proc macros where we can avoid it.

Reply:
Good idea! I am still new to rust macros, but macro_rules seems to make more sense here.